exporters/otlp/common/src/test/java/io/opentelemetry/exporter/internal/TlsUtilTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.TrustManager;
+import org.junit.jupiter.api.Test;
+
+class TlsUtilTest {
+
+  @Test
+  void createSslSocketFactoryTest() throws IOException {
```

Comment:
It's sort of unclear what this is testing, I think we should remove this test and have actual usage of client TLS in our exporter test suite

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractGrpcTelemetryExporterTest.java

We also won't need to check in any of these certificates and can use the selfsignedcertificate extension.

`tlsCustomizer` can be used to add trusted client certificates to the server

https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/server/ServerBuilder.html

If it's unclear how to make the test work I can help by adding to this PR, let me know

Reply:
There are several parts I can see in this comment:

1. unclear test scope - It was an attempt to test also loading PEM file. 
2. I understand the point in not checking in the files at all and replace them with extension for certificates. I thought it might be useful to provide example to users how provided file should look like. It it should be with start/stop block, if it should be binary of base64 encoded. May be it wasn't the best idea. I can remove it / try to replace with tlsCustomizer
3. tlsCustomizer - will test less added code. 
4. I am not sure if got it completely. You want me to extend ServerBuilder with something like: 
`sb.tlsCustomizer(sslContextBuilder -> sslContextBuilder.keyManager(certificate.privateKey(), certificate.certificate()));`
? (should I create another key pair?)  If I am far from needed change in test then may be it would be better if you could help with test.

Reply:
We need to add another selfsignedcertificate for what's used by the TLS-enabled client. 
`static final SelfSignedCertificateExtension clientCertificate = new SelfSignedCertificateExtension();`

We'll need to add that certificate as a trusted one for the server:
`sb.tlsCustomizer(ssl -> ssl.trustManager(clientCertificate.certificate())`
(BTW we don't want to change the server's TLS context as your snippet would do)

And then a test case for client TLS

```java
@Test
void clientTls() {
  exporterBuilder().setClientTls(Files.readAllBytes(clientCertificate.privateKey().toPath()), Files.readAllBytes(clientCertificate.certificate())

  // Similar to other tests
```

I think this will test what we want.

Do these hints help? If not I can add a commit with them.

Reply:
I think it does.
If it is still not what you expected then feel free to stop me and provide tests as you see fit.

Reply:
Thanks! Let's go ahead and remove this test and related files, I think the server test looks swell.

Reply:
done